cmd/agent/app/options/options.go

```diff
@@ -57,6 +57,7 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	}
 
 	fs.BoolVar(&o.LeaderElection.LeaderElect, "leader-elect", true, "Start a leader election client and gain leadership before executing the main loop. Enable this when running replicated components for high availability.")
+	fs.StringVar(&o.LeaderElection.ResourceNamespace, "lock-namespace", util.NamespaceKarmadaSystem, "Define the namespace of resource object that will be used to lock")
```

Comment:
how about use `leader-election-namespace` ?

Reply:
ok, let me update this.

Reply:
`leader-election-namespace` is better than `lock-namespace`. 
But I like [the flags defined](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/) by `kube-controller-manager` better:
- --leader-elect-resource-name
- --leader-elect-resource-namespace
- --leader-elect-resource-lock

It'd be more clear when we add more flags against the `resource`.

@pigletfly How do you think?

Don't get me wrong, I didn't mean to add all the flags above on this PR.

Reply:
sounds good.